contracts/governance/Staking/Staking.sol

```diff
@@ -298,21 +298,108 @@ contract Staking is
-        vestingWhitelist[vesting] = false;
+    function _governanceWithdrawVestingDirect(
+        uint96 amount,
+        uint256 until,
+        address receiver,
+        VestingConfig memory vestingConfig
+    ) internal {
+        _notSameBlockAsStakingCheckpoint(until);
+
+        _withdrawFromTeamVesting(amount, until, receiver, vestingConfig);
+    }
+
+    /**
+     * @notice New governance withdraw vesting directly through staking contract.
+     * This direct withdraw vesting, can solve the out of gas issue from the old withdraw vesting function.
+     * This function only allowing the call from vesting contract which type is TeamVesting.
+     *
+     * @param vesting The vesting address.
+     * @param receiver The receiving address.
+     * @param startFrom The start value for the iterations.
+     */
+    function cancelTeamVesting(
+        address vesting,
+        address receiver,
+        uint256 startFrom
+    ) external onlyAuthorized whenNotFrozen {
+        _cancelTeamVesting(vesting, receiver, startFrom);
+    }
+
+    /**
+     * @notice Withdraws tokens from the staking contract and forwards them
+     * to an address specified by the token owner. Low level function.
+     * @dev Once here the caller permission is taken for granted.
+     * @param _vesting The vesting address.
+     * @param _receiver The receiving address.
+     * @param _startFrom The start value for the iterations.
+     * or just unlocked tokens (false).
+     * */
+    function _cancelTeamVesting(
+        address _vesting,
+        address _receiver,
+        uint256 _startFrom
+    ) private {
+        require(_receiver != address(0), "receiver address invalid");
+
+        ITeamVesting teamVesting = ITeamVesting(_vesting);
+
+        VestingConfig memory vestingConfig =
+            VestingConfig(
+                _vesting,
+                teamVesting.startDate(),
+                teamVesting.endDate(),
+                teamVesting.cliff(),
+                teamVesting.duration(),
+                teamVesting.tokenOwner()
+            );
+
+        uint96 tempStake;
+
+        /// @dev Usually we just need to iterate over the possible dates until now.
+        uint256 end;
+
+        uint256 defaultStart = vestingConfig.startDate + vestingConfig.cliff;
+
+        _startFrom = _startFrom >= defaultStart ? _startFrom : defaultStart;
+
+        /// @dev In the unlikely case that all tokens have been unlocked early,
+        ///   allow to withdraw all of them.
+        end = vestingConfig.endDate;
 
-        emit VestingTokensWithdrawn(vesting, receiver);
+        uint256 totalIterationValue =
+            (_startFrom + (TWO_WEEKS * getMaxVestingWithdrawIterations()));
+        uint256 adjustedEnd = end < totalIterationValue ? end : totalIterationValue;
+
+        /// @dev Withdraw for each unlocked position.
+        for (uint256 i = _startFrom; i < adjustedEnd; i += TWO_WEEKS) {
```

Comment:
```suggestion
for (uint256 i = _startFrom; i <= adjustedEnd; i += TWO_WEEKS)
```
it used to be `<= end` 
why is it now `<` - was it a bug previously? could you justify the change?